src/Atmos/Parameterizations/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -45,146 +42,165 @@ module SurfaceFluxes
-  ψ_h = 2 * log1p((y(ζ) - y(ζ_0))/(1 + y(ζ_0))) # log((1 + y(ζ))/(1 + y(ζ_0)))
-  return ψ_h
+function Base.show(io::IO, sfc::SurfaceFluxConditions)
+  println(io, "----------------------- SurfaceFluxConditions")
+  println(io, "momentum_flux           = ", sfc.momentum_flux)
+  println(io, "buoyancy_flux           = ", sfc.buoyancy_flux)
+  println(io, "Monin_Obukhov_length    = ", sfc.Monin_Obukhov_length)
+  println(io, "friction_velocity       = ", sfc.friction_velocity)
+  println(io, "temperature_scale       = ", sfc.temperature_scale)
+  println(io, "pottemp_flux            = ", sfc.pottemp_flux)
+  println(io, "exchange_coeff_momentum = ", sfc.exchange_coeff_momentum)
+  println(io, "exchange_coeff_heat     = ", sfc.exchange_coeff_heat)
+  println(io, "-----------------------")
 end
 
 """
-    compute_MO_len(u, flux)
-
-Computes the Monin-Obukhov length (Eq. 3 Ref. Byun1990)
-"""
-compute_MO_len(u, flux) = - u^3 / (flux * k_Karman)
-
+    surface_conditions(u_ave::DT,
+                       θ_bar::DT,
+                       θ_ave::DT,
+                       θ_s::DT,
+                       Δz::DT,
+                       z_0_m::DT,
+                       z_0_h::DT,
+                       z::DT,
+                       F_m::DT,
+                       F_h::DT,
+                       a::DT,
+                       Pr::DT,
+                       pottemp_flux::DT
+                       ) where DT<:AbstractFloat
+
+Surface conditions given
+ - `u_ave` volume-averaged horizontal wind speed
+ - `θ_ave` volume-averaged potential temperature
+ - `θ_bar` basic potential temperature
+ - `θ_s` surface potential temperature
+ - `Δz` layer thickness (not spatial discretization)
+ - `z_0_m` roughness length for momentum
+ - `z_0_h` roughness length for heat
+ - `z` coordinate axis
+ - `F_m` momentum flux at the top
+ - `F_h` heat flux at the top
+ - `a` free model parameter with prescribed value of 4.7
+ - `Pr` Prantl number at neutral stratification
+ - `pottemp_flux` potential temperature flux (optional)
+
+If `pottemp_flux` is not given, then it is computed by iteration
+of equations 3, 17, and 18 in Nishizawa2018.
 """
-    compute_friction_velocity(u_ave, flux, z_0, z_1, β_m, γ_m, tol_abs, iter_max)
+function surface_conditions(u_ave::DT,
+                            θ_bar::DT,
+                            θ_ave::DT,
+                            θ_s::DT,
+                            Δz::DT,
+                            z_0_m::DT,
+                            z_0_h::DT,
+                            z::DT,
+                            F_m::DT,
+                            F_h::DT,
+                            a::DT,
+                            Pr::DT,
+                            pottemp_flux::Union{Nothing, DT}=nothing
+                            ) where DT<:AbstractFloat
+
+  L_MO = 1.0
+  if pottemp_flux!=nothing
+
+    @assert 0 <= z <= z_0_m
+    @assert 0 <= z <= z_0_h
+
+    xatol, maxiters = DT(1e-3), 100
```

Comment:
Do we really need so high a default ‘maxiters’?

Reply:
No, I don't think so, I can adjust these if we don't end up removing this case (when `pottemp_flux` is passed as an input argument).

Reply:
Having `pottemp_flux` as optional input would be good for now. We'll need it for many experiments.

Reply:
Ok, I managed to allow both cases using the same code base, but we'll need to re-work some things to get this working on the GPU (also, I don't know if NLSolve is GPU friendly).